dev/conformance/conformance-tests/set-mergeall-empty.json

```diff
@@ -16,9 +16,6 @@
               "update": {
                 "name": "projects/projectID/databases/(default)/documents/C/d",
                 "fields": {}
-              },
```

Comment:
Do you know why this had to change? It feels like what you have is correct, but I am a bit concerned that it changed  in your branch. 

If this change is correct, we need to update the description.

Reply:
You can see the diff responsible for this [here](https://github.com/googleapis/nodejs-firestore/compare/master...bc/bulk-master?expand=1#diff-9251327f0da3d89ce43887319402b935R347).

In the original logic: if the `merge` option is specified and the `transform` field is empty, add the `updateMask` field. This meant that we could have an empty `updateMask` field as long as the `transform` field was empty.

With the changes to `updateTransform`, I changed the code to only add the updateMask if it wasn't empty. To preserve the original behavior, we can change the line of code I mentioned above to:
```
if (hasMerge && (!documentMask.isEmpty || transform.isEmpty))
```

Does including an empty `updateMask` change behavior? If not, I think it'd be better to leave the field out. The description is still valid, since we are still checking that a merge with empty data results in a valid request being created.

Reply:
Actually, come to think of this, this is now broken. Without the update mask, set({}, {merge:true}) is now the same as set({}, {merge:true}) and replaces all data on the backend. Instead, it should preserve the document on the backend and be a no op if the document already exists.

Can you fix this? I will look at the rest of this diff once this is fixed.